File: internal/manager/agent/paths.go
// Package agent detects supported agents and inspects their configuration.
package agent

import "path/filepath"

// Kind identifies a supported agent.
type Kind string

const (
	ClaudeCode Kind = "claude"
	OpenCode   Kind = "opencode"
	Codex      Kind = "codex"
)

// Paths contains the files used to configure an agent. AuthPath is set only
// for agents that keep authentication separately from their main config.
type Paths struct {
	ConfigPath string
	AuthPath   string
}

// ClaudePaths applies the CLAUDE_CONFIG_DIR semantics used by the setup
// scripts. An empty override selects the default under the user's home.
func ClaudePaths(home, configDir string) Paths {
	if configDir == "" {
		configDir = filepath.Join(home, ".claude")
	}
	return Paths{ConfigPath: filepath.Join(configDir, "settings.json")}
}

// OpenCodePaths applies the OPENCODE_CONFIG and JSON-before-JSONC fallback
// semantics used by the setup scripts.
func OpenCodePaths(home, configOverride string) Paths {
	if configOverride != "" {
		return Paths{ConfigPath: configOverride}
	}

	dir := filepath.Join(home, ".config", "opencode")
	jsonPath := filepath.Join(dir, "opencode.json")
	jsoncPath := filepath.Join(dir, "opencode.jsonc")
	if isRegularFile(jsonPath) {
		return Paths{ConfigPath: jsonPath}
	}
	if isRegularFile(jsoncPath) {
		return Paths{ConfigPath: jsoncPath}
	}
	return Paths{ConfigPath: jsonPath}
}

// CodexPaths applies CODEX_HOME semantics used by both setup scripts.
func CodexPaths(home, codexHome string) Paths {
	if codexHome == "" {
		codexHome = filepath.Join(home, ".codex")
	}
	return Paths{
		ConfigPath: filepath.Join(codexHome, "config.toml"),
		AuthPath:   filepath.Join(codexHome, "auth.json"),
	}
}

File: desktop/src-tauri/src/manager_core/agent/paths.rs
use std::path::{Path, PathBuf};

#[derive(Clone, Debug, Eq, PartialEq)]
pub struct Paths {
    pub config_path: PathBuf,
    pub auth_path: Option<PathBuf>,
}

pub fn claude_paths(home: &Path, config_dir: &str) -> Paths {
    let dir = if config_dir.is_empty() {
        home.join(".claude")
    } else {
        PathBuf::from(config_dir)
    };
    Paths {
        config_path: dir.join("settings.json"),
        auth_path: None,
    }
}

pub fn opencode_paths(home: &Path, config_override: &str) -> Paths {
    if !config_override.is_empty() {
        return Paths {
            config_path: PathBuf::from(config_override),
            auth_path: None,
        };
    }
    let dir = home.join(".config").join("opencode");
    let json_path = dir.join("opencode.json");
    let jsonc_path = dir.join("opencode.jsonc");
    if is_regular_file(&json_path) {
        return Paths {
            config_path: json_path,
            auth_path: None,
        };
    }
    if is_regular_file(&jsonc_path) {
        return Paths {
            config_path: jsonc_path,
            auth_path: None,
        };
    }
    Paths {
        config_path: json_path,
        auth_path: None,
    }
}

pub fn codex_paths(home: &Path, codex_home: &str) -> Paths {
    let dir = if codex_home.is_empty() {
        home.join(".codex")
    } else {
        PathBuf::from(codex_home)
    };
    Paths {
        config_path: dir.join("config.toml"),
        auth_path: Some(dir.join("auth.json")),
    }
}

pub fn is_regular_file(path: &Path) -> bool {
    std::fs::metadata(path)
        .map(|info| info.is_file())
        .unwrap_or(false)
}
